src/main/java/io/supertokens/thirdparty/ThirdParty.java

```diff
@@ -158,6 +162,18 @@ public static UserPaginationContainer getUsers(Main main,
         return new UserPaginationContainer(resultUsers, nextPaginationToken);
     }
 
+    public static UserInfo[] getUsersByEmail(Main main, String email) throws StorageQueryException, InvalidInputException {
+        if (email == null || email.equals("")) {
```

Comment:
- As per our current API design, validation of input happens in the API and not here.. But I'm OK with this. 
- If you want email to not be null, you should mark the input param as @Nonnull and not have to check here again.
- Also, if you are checking `EmailValidator.isValid` below, why do you need to check if it's not an empty string here?

Reply:
> * As per our current API design, validation of input happens in the API and not here.. But I'm OK with this.

Good, I'm certain that validation not happening in the API layer is a good design choice (isolation from API layer changes and reduced duplication).

> * If you want email to not be null, you should mark the input param as @nonnull and not have to check here again.

That's why there's `@Nonnull` on storage layer parameter. I don't want to move this bit of validation logic "up", to an entity that calls `getUsersByEmail`. This would lead to unnecessary duplication.

> * Also, if you are checking `EmailValidator.isValid` below, why do you need to check if it's not an empty string here?

I will move that to `EmailValidatior.isValid`, it makes more sense.